dnsapi/dns_schlundtech.sh

```diff
@@ -0,0 +1,200 @@
+  echo "$_SLTEC_response" | grep "<code>S0202</code>" >/dev/null
+  _ST_rm_result="$?"
+  _debug "result: ${_ST_rm_result}"
+
+  # no return value documented
+  #return "$_ST_rm_result"
+}
+
+####################  private functions below ##################################
+
+_SLTEC_credentials() {
+
+  if [ -z "${SLTEC_context}" ]; then
+    SLTEC_context="${SLTEC_context_default}"
+  fi
+
+  if [ -z "${SLTEC_server}" ]; then
+    SLTEC_server="${SLTEC_server_default}"
+  fi
+
+  if [ -z "${SLTEC_user}" ] || [ -z "$SLTEC_password" ] || [ -z "${SLTEC_context}" ] || [ -z "${SLTEC_server}" ]; then
+    SLTEC_user=""
+    SLTEC_password=""
+    SLTEC_context=""
+    SLTEC_server=""
+    return 1
+  else
+    _saveaccountconf SLTEC_user "${SLTEC_user}"
+    _saveaccountconf SLTEC_password "${SLTEC_password}"
+    _saveaccountconf SLTEC_context "${SLTEC_context}"
+    _saveaccountconf SLTEC_server "${SLTEC_server}"
+    return 0
+  fi
+}
+
+_SLTEC_split_domain() {
+  _ST_split_fulldomain="$1"
+
+  _SLTEC_domain="$(echo "$_ST_split_fulldomain" | sed 's/.*\.\([^.]*\.[^.]*\)/\1/')"
+  _SLTEC_subdomain="$(echo "$_ST_split_fulldomain" | sed 's/\(.*\)\.[^.]*\.[^.]*/\1/')"
+}
+
+_SLTEC_init_request_add() {
+  _ST_init_add_user="$1"
+  _ST_init_add_password="$2"
+  _ST_init_add_context="$3"
+  _ST_init_add_domain="$4"
+  _ST_init_add_subdomain="$5"
+  _ST_init_add_value="$6"
+
+  _SLTEC_xmladd="<?xml version='1.0' encoding='utf-8'?>
+  <request>
+    <auth>
+      <user>${_ST_init_add_user}</user>
+      <password>${_ST_init_add_password}</password>
+      <context>${_ST_init_add_context}</context>
+    </auth>
+    <task>
+      <code>0202001</code>
+      <default>
+        <rr_add>
+          <name>${_ST_init_add_subdomain}</name>
+          <type>TXT</type>
+          <value>${_ST_init_add_value}</value>
+          <ttl>60</ttl>
+        </rr_add>
+      </default>
+      <zone>
+        <name>${_ST_init_add_domain}</name>
+      </zone>
+    </task>
+  </request>"
+}
+
+_SLTEC_init_request_rm() {
+  _ST_init_rm_user="$1"
+  _ST_init_rm_password="$2"
+  _ST_init_rm_context="$3"
+  _ST_init_rm_domain="$4"
+  _ST_init_rm_subdomain="$5"
+  _ST_init_rm_value="$6"
+
+  _SLTEC_xmlrm="<?xml version='1.0' encoding='utf-8'?>
+  <request>
+    <auth>
+      <user>${_ST_init_rm_user}</user>
+      <password>${_ST_init_rm_password}</password>
+      <context>${_ST_init_rm_context}</context>
+    </auth>
+    <task>
+      <code>0202001</code>
+      <default>
+        <rr_rem>
+          <name>${_ST_init_rm_subdomain}</name>
+          <type>TXT</type>
+          <value>${_ST_init_rm_value}</value>
+        </rr_rem>
+      </default>
+      <zone>
+        <name>${_ST_init_rm_domain}</name>
+      </zone>
+    </task>
+  </request>"
+}
+
+_SLTEC_send_request() {
+  _ST_send_request="$1"
+  _ST_send_url="$2"
+
+  _SLTEC_response="$(curl -s -H "Content-type: text/xml" --data-binary "${_ST_send_request}" "${_ST_send_url}")"
```

Comment:
do not use `curl` directly here.
please use `_get()` or `_post()` function instead.

Reply:
Thanx, I checked that out and dug into the curl function. The `_post` function, when using `curl`, sends data with the `--data` switch. According to the manpage that's the same as `--data-ascii`. This is **not** the same as using the `--data-binary` switch. Unfortunately the curl manpage is not very precise about it. Actually it's written in the desciption of `--data-binary`: `Data is posted in a similar manner as --data-ascii does, except that newlines are preserved and conversions are never done.` This implies that `--data` removes newlines which in fact it does. I just checked it out with a small php program. IMHO the `_post` function should use the `--data-binary` switch to ensure that the data given is passed to the server as is, without changes. Only the caller knows exactly which format is needed. The `_post` function should pass that data on to the server transparently.
I'll create a ticket, just for the records.